content/blog/2020-11-11-self/index.md

```diff
@@ -0,0 +1,155 @@
++++
+title = "An Efficient Implementation of SELF, a Dynamically-Typed Object-Oriented Language Based on Prototypes"
+[extra]
+bio = """
+Evan Adler is a Masters of Engineering student at Cornell interested in compilers and programming languages. 
+"""
+[[extra.authors]]
+name = "Evan Adler"
+latex = true
++++
+
+## Background
+
+This week, we discuss the paper [An Efficient Implementation of SELF, a Dynamically-Typed Object-Oriented Language Based on Prototypes][efficientself] (Chambers, Ungar, and Lee), which appeared in OOPSLA 1989. As it turns out, the same year which heralded the downfall of communism in Europe also marked a significant victory in the liberation of object-oriented programming languages from class-based domination. What I mean is, as we'll see, Self is the first object-oriented programming language without classes, something not really considered possible before its debut.
+
+Over the years Self has been a research endeavour at three of the most important institutions in the history of Computer Science: Xerox PARC, Stanford University, and Sun Microsystems. As far as I can tell, this week's paper is the fourth Self paper to be published. The first, [SELF: The Power of Simplicity][self] (Ungar and Smith), written by the language's creators, introduced Self in 1987. The second, [Object Storage and Inheritance for Self][leethesis] (Lee), was written in 1988 and discusses Self's object layout (at that point in time) and Self's multiple inheritance rules. The third, [Customization: Optimizing Compiler Technology for Self, a Dynamically-Typed Object-Oriented Programming Language][customization] (Chambers and Ungar), written earlier in 1989, covers some of the same optimizations discussed in the Chambers, Ungar, and Lee paper.
```

Comment:
That's why I put it in the Background section. I think I included it because my intuition is to understand the language first, and then understand the implementation. But you're definitely right that this detail is distracting.